File: static_arrays/26_Remove_Duplicate/26_remove_duplicated_from_sorted_array.go
package main

func main() {
	nums := []int{0, 1, 2, 2, 3, 0, 4, 2}
	println(removeDuplicates(nums))
}
func removeDuplicates(nums []int) int {
	length := len(nums)
	if length < 2 {
		return length
	}
	k, i := 1, 1
	for i < length {
		if nums[i] != nums[i-1] {
			nums[k] = nums[i]
			k++
		}
		i++
	}
	return k
}

File: static_arrays/26_Remove_Duplicate/26_remove_duplicated_from_sorted_array.js
/**
 * Linear 
 * Time O(N) | Space O(1)
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = (nums) => {
    let [left, right] = [0, 0];

    while (right < nums.length) {
        const [leftVal, rightVal] = [nums[left], nums[right]];

        const isEqual = (rightVal === leftVal);
        if (!isEqual) {
            left++;
            nums[left] = rightVal;
        }

        right++;
    }

    return (left + 1);
};

var removeDuplicates = function (nums) {
    let [left, right] = [0, 0];

    for (right = 1; right < nums.length; right++) {
        const [leftVal, rightVal] = [nums[left], nums[right]]
        if (rightVal !== leftVal) {
            left++;
            nums[left] = rightVal;
        }
    }
    return {
        name: (left + 1),
        nums
    };
};

var removeDuplicates = (nums) => {
    let [left, right] = [0, 0];

    while (right < nums.length) {
        const [leftVal, rightVal] = [nums[left], nums[right]];

        const isEqual = (rightVal === leftVal);
        if (!isEqual) {
            left++;
            nums[left] = rightVal;
        }

        right++;
    }

    return (left + 1);
};
